### PR TITLE
security topic sns subscription

### PR DESCRIPTION
**Pull request by banovicn**
This is PR to support subscribing to SNS topic, **aws-controltower-AggregateSecurityNotifications**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -98,6 +98,21 @@ variable "monitor_iam_access" {
...
+
+variable "sns_endpoint_protocol" {
+  type        = string
+  description = "Endpoint protocol for SNS topic subscription"
+}
+
+variable "sns_security_topic_subscription" {
+  type        = bool
+  description = "Enable SNS aggregated security topic subscription"
+}
```
Comment (shoekstra) [inline variables.tf:114]: Could these values be specified in a single `sns_security_subscription` object?

File: modules/security_hub/variables.tf [hunk 2]
```diff
@@ -15,3 +20,19 @@ variable "region" {
...
+variable "sns_endpoint_protocol" {
+  type        = string
+  description = "Endpoint protocol for SNS topic subscription"
+}
+
+variable "sns_security_topic_subscription" {
+  type        = bool
+  default     = false
+  description = "Enable SNS aggregated security topic subscription"
+}
```
Comment (shoekstra) [inline modules/security_hub/variables.tf:38]: Could these values be specified in a single `sns_security_subscription` object?

File: modules/security_hub/variables.tf [hunk 3]
```diff
@@ -1,3 +1,8 @@
+variable "account_id" {
+  type        = string
+  description = "AWS Account ID"
+}
```
Comment (shoekstra) [inline modules/security_hub/variables.tf:4]: What is this for? There is no reference to this variable anywhere else.

File: README.md [hunk 4]
```diff
@@ -139,6 +139,11 @@ module "landing_zone" {
   ]
 ```
 
+### Enable SNS topic subscription
+
+If you would like to subscribe to aggregated security SNS topic created by Control Tower, set `sns_security_topic_subscription` variable to `true`.
+And provide values for your endpoint to receive notifications, variable `sns_endpoint` and protocol to be used, variable `sns_endpoint_protocol`.
```
Comment (shoekstra) [inline README.md:145]: This needs to be reworded as it's hard to follow in the current form. Can you please also add an example like other opt-in settings.

**Review by shoekstra — CHANGES_REQUESTED** [review]

File: modules/security_hub/variables.tf (hunk 3 above)
Comment (banovicn) [reply modules/security_hub/variables.tf:4]: it is referenced in `topic_arn  = "arn:aws:sns:${var.region}:${var.account_id}:aws-controltower-AggregateSecurityNotifications"`, **line 34** in **modules/security_hub/main.tf**

File: README.md (hunk 4 above)
Comment (banovicn) [reply README.md]: done

File: modules/security_hub/variables.tf (hunk 2 above)
Comment (banovicn) [reply modules/security_hub/variables.tf]: done

File: variables.tf (hunk 1 above)
Comment (banovicn) [reply variables.tf]: done

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,7 +4,11 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/) and this project adheres to [Semantic Versioning](http://semver.org/).
 
-## Unreleased
+## Unreleased (2020-12-29)
```
Comment (shoekstra) [inline CHANGELOG.md:7]:
```suggestion
## Unreleased
```

No need to append a date to the unreleased header.

File: modules/security_hub/main.tf [hunk 6]
```diff
@@ -26,3 +26,11 @@ resource "aws_securityhub_standards_subscription" "default" {
   standards_arn = each.value
   depends_on    = [aws_securityhub_account.default]
 }
+
+resource "aws_sns_topic_subscription" "datadog-security" {
```
Comment (shoekstra) [inline modules/security_hub/main.tf:30]:
```suggestion
resource "aws_sns_topic_subscription" "datadog_security" {
```

We should use `_` in resource names.

File: modules/security_hub/variables.tf [hunk 7]
```diff
@@ -1,3 +1,9 @@
+variable "account_id" {
+  type        = string
+  default     = null
+  description = "AWS Account ID"
```
Comment (shoekstra) [inline modules/security_hub/variables.tf:4]: What AWS Account ID should be entered here? The description could do with some updating 🙂

File: modules/security_hub/variables.tf [hunk 8]
```diff
@@ -15,3 +21,12 @@ variable "region" {
   default     = "eu-west-1"
   description = "The name of the AWS region where SecurityHub will be enabled"
 }
+
+variable "sns_security_subscription" {
+  type = list(object({
+    sns_endpoint          = string
+    sns_endpoint_protocol = string
```
Comment (shoekstra) [inline modules/security_hub/variables.tf:28]:
```suggestion
    endpoint = string
    protocol = string
```

No need to include an `sns` prefix if the variable is about SNS already.

Does it make sense to trim `endpoint_protocol` down to just `protocol` too?

File: variables.tf [hunk 9]
```diff
@@ -101,6 +101,15 @@ variable "monitor_iam_access" {
   }
 }
 
+variable "sns_security_subscription" {
+  type = list(object({
+    sns_endpoint          = string
+    sns_endpoint_protocol = string
```
Comment (shoekstra) [inline variables.tf:107]:
```suggestion
    endpoint = string
    protocol = string
```

No need to include an `sns` prefix if the variable is about SNS already.

Does it make sense to trim `endpoint_protocol` down to just `protocol` too?

File: README.md [hunk 10]
```diff
@@ -148,6 +148,22 @@ module "landing_zone" {
...
+If you need to subscribe to AggregatedSecurityNotifications topic in order to receive security findings, please set values for `sns_endpoint` and `sns_endpoint_protocol` variables.
+
+Example for https protocol and specified webhook endpoint:
+
+```hcl
+module "landing_zone"{
+  ...
+  
+  sns_endpoint          = "https://app.datadoghq.com/intake/webhook/sns?api_key=qwerty0123456789"
+  sns_endpoint_protocol = "https"
```
Comment (shoekstra) [inline README.md:162]:
```suggestion
  sns_security_subscription = {
    endpoint = "https://app.datadoghq.com/intake/webhook/sns?api_key=qwerty0123456789"
    protocol = "https"
  }
```

**Review by shoekstra — CHANGES_REQUESTED** [review]

**Review by shoekstra — CHANGES_REQUESTED** [review]

File: audit.tf [hunk 11]
```diff
@@ -1,3 +1,11 @@
+locals {
+  sns_security_subscription = [
+    for sub in var.sns_security_subscription :
+    merge(sub, { account_id = var.control_tower_account_ids.audit }
+    )
```
Comment (shoekstra) [inline audit.tf:5]:
```suggestion
    merge(sub, { account_id = var.control_tower_account_ids.audit })
```

**Review by shoekstra — CHANGES_REQUESTED** [review]

**Review by shoekstra — APPROVED** [review]
